packages/flutter_driver/lib/src/driver.dart

```diff
@@ -94,7 +97,9 @@ class FlutterDriver {
   /// Creates a driver that uses a connection provided by the given
   /// [_serviceClient], [_peer] and [_appIsolate].
   @visibleForTesting
-  FlutterDriver.connectedTo(this._serviceClient, this._peer, this._appIsolate);
+  FlutterDriver.connectedTo(this._serviceClient, this._peer, this._appIsolate,
+      { bool printCommunication: false, bool logCommunicationToFile: true })
+      : _printCommunication = printCommunication, _logCommunicationToFile = logCommunicationToFile;
```

Comment:
How about instead of adding these options we publish a "stream" of log messages and provide utility functions to log them to stdout and files? This way the driver won't have to decide where the log messages go. I think producing log messages and logging them are separate concerns.

One use-case where this could be problematic is in multi-device testing. There you have multiple instances of `FlutterDriver`, each talking to its own device. If we have them write to the same file, things can get messy.